frontend/main.py

```diff
@@ -160,19 +160,21 @@ def get_queues():
 def resolve_correction(queue_name, correction_id):
     data = request.json
     if "action" not in data or (
-        data["action"] == "approve" and "corrected_val" not in data
+        data["action"] == "approve" and "corrected_val" not in data and "corrected_grade" not in data
     ):
         return jsonify(
             create_response_message(
                 False,
-                'Please provide valid action param "reject", "skip", or "approve" and "corrected_val" param',
+                'Please provide valid action param "reject", "skip", or "approve" and "corrected_val" or "corrected_grade" param',
             )
         )
 
     action = data["action"]
     email_template = data.get("emailTemplate", "")
     version = data.get("version", 0)
     username = request.cookies.get("username")
+    corrected_val = data.get("corrected_val", "")
+    corrected_grade = data.get("corrected_grade", "")
```

Comment:
instead of corrected_grade, we need to capture the attribute that was modified which can be either "hadithText" or "grade"

Reply:
did you mean the attribute should be renamed to hadithText and grade, instead of corrected_val and corrected_grade?

Reply:
no. We'll only allow one attribute to be changed per correction.